components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/action/PreIssueAccessTokenResponseProcessor.java

```diff
@@ -154,14 +163,58 @@ private void logOperationExecutionResults(ActionType actionType,
     }
 
     @Override
-    public ActionExecutionStatus processErrorResponse(Map<String, Object> map, Event event,
-                                                      ActionInvocationErrorResponse actionInvocationErrorResponse)
+    public ActionExecutionStatus<Failure> processFailureResponse(Map<String, Object> eventContext,
+                                                        Event actionEvent,
+                                                        ActionInvocationFailureResponse failureResponse) throws
+            ActionExecutionResponseProcessorException {
+
+        // This throws an exception as server_error is expected to be handled under ERROR status.
+        handleInvalidErrorCodes(failureResponse.getFailureReason());
+        return new FailedStatus(new Failure(failureResponse.getFailureReason(),
+                failureResponse.getFailureDescription()));
+    }
+
+    private void handleInvalidErrorCodes(String errorCode) throws ActionExecutionResponseProcessorException {
+
+        if (isServerError(errorCode)) {
+            if (LoggerUtils.isDiagnosticLogsEnabled()) {
+                DiagnosticLog.DiagnosticLogBuilder diagLogBuilder = new DiagnosticLog.DiagnosticLogBuilder(
+                        ActionExecutionLogConstants.ACTION_EXECUTION_COMPONENT_ID,
+                        ActionExecutionLogConstants.ActionIDs.VALIDATE_ACTION_OPERATIONS);
+                diagLogBuilder
+                        .resultMessage("Internal server errors must be sent with an ERROR status, not a FAILED status")
+                        .logDetailLevel(DiagnosticLog.LogDetailLevel.APPLICATION)
+                        .resultStatus(DiagnosticLog.ResultStatus.FAILED)
+                        .build();
+                LoggerUtils.triggerDiagnosticLogEvent(diagLogBuilder);
```

Comment:
These constants are now changed with https://github.com/wso2/carbon-identity-framework/pull/6093/files
Use the appropriate.

Reply:
Also, refactor the message.
Mention something as ""server_error" is not allowed for "failureReason" attribute in Failed response" which conveys the message clearly

Reply:
Or we can just say "Invalid value for "failedReason" attribute at FAILED state." Then we can mention what is not accepted and what's accepted in docs

Reply:
This is updated with ad8eab9323eceb34d330062422c6f893a2a58b9f